src/pages/Books/Books.tsx

```diff
@@ -38,6 +41,11 @@ export const Books: React.FC = () => {
     }
 
     if (searchQuery === '') {
+      setMessage(
+        '📚 Having a bookworm memory block? Type anything to get started!',
+      );
+      setBooks([]);
```

Comment:
This is basically this -> https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

You are reacting to the user's action when he removes everything. This should be moved as close to the action handling as possible. We have kept the fetching in the useEffect just because of the setTimeout debouncing you did. It was just easier to keep there (and I was too lazy to think about other out-of-useEffect approach).

Right now, you are passing `setSearchQuery` into the input. Change that prop name to `handleSearchUpdate` (react docs mention prefixing handlers with `on` -> https://react.dev/learn/responding-to-events#adding-event-handlers, I prefer `handleX` when I am supposed to implement the whole handler, and `onX` when I am just reacting to the click and most of the handling is done for me. Sorry, too deep)

The `handleSearchUpdate` will be declared here and passed via a prop. The function will update the setSearchQuery. update the books and update the message. 

The useEffect should not know about it.